packages/rxdart_flutter/lib/src/value_stream_listener.dart

```diff
@@ -0,0 +1,174 @@
+///
+/// **Example**
+///
+/// ```dart
+/// ValueStreamListener<T>(
+///   stream: valueStream,
+///   listener: (context, previous, current) {
+///     // do stuff here based on valueStream's
+///     // previous and current values
+///   },
+///   child: Container(),
+/// )
+/// ```
+/// {@endtemplate}
+class ValueStreamListener<T> extends StatefulWidget {
+  /// {@macro value_stream_listener}
+  const ValueStreamListener({
+    Key? key,
+    required this.stream,
+    required this.listener,
+    required this.child,
+    this.isReplayValueStream = true,
+  }) : super(key: key);
+
+  /// The [ValueStream] that the [ValueStreamConsumer] will interact with.
+  final ValueStream<T> stream;
+
+  /// Takes the `BuildContext` along with the `previous` and `current` values
+  ///  and is responsible for executing in response to `value` changes.
+  final ValueStreamWidgetListener<T> listener;
+
+  /// The widget which will be rendered as a descendant of the
+  /// [ValueStreamListener].
+  final Widget child;
+
+  /// Whether or not the [stream] emits the last value
+  /// like [BehaviorSubject] does.
+  ///
+  /// Defaults to `true`.
+  final bool isReplayValueStream;
+
+  @override
+  State<ValueStreamListener<T>> createState() => _ValueStreamListenerState<T>();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty<ValueStream<T>>('stream', stream))
+      ..add(
+          DiagnosticsProperty<bool>('isReplayValueStream', isReplayValueStream))
+      ..add(ObjectFlagProperty<ValueStreamWidgetListener<T>>.has(
+          'listener', listener))
+      ..add(ObjectFlagProperty<Widget>.has('child', child));
+  }
+}
+
+class _ValueStreamListenerState<T> extends State<ValueStreamListener<T>> {
+  StreamSubscription<T>? _subscription;
+  late T _currentValue;
+  ErrorAndStackTrace? _error;
+  bool _initialized = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _subscribe();
+    _initialized = true;
+  }
+
+  @override
+  void didUpdateWidget(covariant ValueStreamListener<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.stream != oldWidget.stream) {
+      _unsubscribe();
+      _subscribe();
+    }
+  }
+
+  @override
+  void dispose() {
+    _unsubscribe();
+    super.dispose();
+  }
+
+  void _subscribe() {
+    final stream = widget.stream;
+
+    _error = validateValueStreamInitialValue(stream);
+    if (_error != null) {
+      return;
+    }
+
+    if (!_initialized) {
+      _currentValue = stream.value;
+    }
+
+    final int skipCount;
+
+    if (widget.isReplayValueStream) {
+      skipCount = _initialized ? 0 : 1;
+    } else {
+      skipCount = 0;
+      if (_initialized) {
+        WidgetsBinding.instance.addPostFrameCallback((_) {
+          _notifyListener(stream.value);
+        });
+      }
+    }
+
+    _subscription = stream.skip(skipCount).listen(
```

Comment:
if skipCount <= 0, don't need to create another Stream 🙏 . 
```suggestion
    _subscription = (skipCount > 0 ? stream.skip(skipCount) : stream).listen(
```